llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -575,6 +578,8 @@ PassBuilder::buildO1FunctionSimplificationPipeline(OptimizationLevel Level,
       SimplifyCFGPass(SimplifyCFGOptions().convertSwitchRangeToICmp(true)));
   FPM.addPass(InstCombinePass());
   invokePeepholeEPCallbacks(FPM, Level);
+  if (SYCLOptimizationMode)
```

Comment:
It would be interesting if non sycl-opt mode can use the pass as well. It seems not possible at the moment as libspirv ControlBarrier implementation is inlined before optimization pass pipeline starts.

Reply:
I probably misunderstanding what SYCLOptimizationMode is. I though it basically mean [O1, O2, O3] on the device code in the frontend (aka no optnone passed to BE), or it's not the case?

Reply:
Moved the pass from here.